### ISSUE TITLE
No ARM image on owncloud/server.

### ISSUE THREAD

**Issue by InnovativeInventor** [owncloud-docker/server#78]
After reading the message on the owncloud official repository in Docker Hub, I realized that even though it was deprecated, it is the only option for ARM users. Is there a plan to support ARM?

**Comment by tboerger** [comment]
We still don't have arm workers for our ci system and I don't know if we will get some in the near future.

First of all we got to wait for drone ci 0.9.

**Comment by PVince81** [comment]
Bumped into this as well yesterday while trying to setup on a QNAP NAS.

We ended up using docker's own ownCloud instance which did boot on ARM but has no DB (sqlite...): https://hub.docker.com/_/owncloud/

**Comment by tboerger** [comment]
@pvince81 but the docker Inc image is deprecated and it's not maintained by us.

@michaelstingl it's not about the machines, it's about the drone integration. I will setup something when we are switching to drone 0.9, than we are able to provide official arm and arm64 images.

**Comment by PVince81** [comment]
Note that we also did not find MariaDB images compatible with ARM. The webhippie one did not run when I tried. So even if we make the OC image itself compatible, the suggested docker compose setup will not run on ARM.

**Comment by tboerger** [comment]
You can use the official mariadb image in that case.

We don't mention arm or arm64 support at all for our docker images, so currently it's up to the user to get a working setup.

**Comment by tboerger** [comment]
And I know that the webhippie images don't provide arm support, they are built by me 😂

**Comment by PVince81** [comment]
the official mariadb image did not boot on arm, that's the problem

**Comment by PVince81** [comment]
neither does the official mysql as maintained by the Oracle team...

**Comment by tboerger** [comment]
But the official mariadb and mysql images got multi arch manifests. But this is anyway offtopic

**Comment by patrickjahns** [comment]
@tboerger 
Shall we move to building the container images for arm on public drone.io ? Or do we want to wait until we have arm runners inside our drone instance?

**Comment by tboerger** [comment]
Maybe we could just move the entire owncloud-docker and owncloud-ci building to the Drone cloud, than we won't have to manage arm or arm64 agents at all.

**Comment by jakutis** [comment]
what's up?

**Comment by tboerger** [comment]
As there had been no update nothing changed so far.

**Comment by jakutis** [comment]
@tboerger, thank you for replying!

After doing these changes in https://github.com/owncloud-docker repositories:

server:
```diff
diff --git a/Dockerfile b/Dockerfile
index c177fd5..1c8b7f5 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM owncloud/base:bionic
+FROM owncloud/base:vytas
 
 LABEL maintainer="ownCloud DevOps <devops@owncloud.com>" \
   org.label-schema.name="ownCloud Server" \
@@ -11,3 +11,13 @@ COPY rootfs /
 
 RUN find /var/www/owncloud \( \! -user www-data -o \! -group root \) -print0 | xargs -r -0 chown www-data:root && \
   chmod g+w /var/www/owncloud
+
+RUN apt-get update -y && \
+  apt-get upgrade -y && \
+  apt-get install -y \
+    wait-for-it
+
+RUN curl -L https://github.com/hairyhenderson/gomplate/releases/download/v3.0.0/gomplate_linux-arm64-slim > /usr/bin/gomplate
+RUN chmod a+rx /usr/bin/gomplate
+RUN curl -L https://github.com/tianon/gosu/releases/download/1.10/gosu-arm64 > /usr/bin/su-exec
+RUN chmod a+rx /usr/bin/su-exec
```

base:
```diff
diff --git a/Dockerfile b/Dockerfile
index d3dc8a2..24b7b0d 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM owncloud/php:latest
+FROM owncloud/php:vytas
 
 LABEL maintainer="ownCloud DevOps <devops@owncloud.com>" \
   org.label-schema.name="ownCloud Base" \
```

php:
```diff
diff --git a/Dockerfile b/Dockerfile
index 4062ddb..44b14d8 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM owncloud/ubuntu:latest
+FROM owncloud/ubuntu:vytas
 
 LABEL maintainer="ownCloud DevOps <devops@owncloud.com>" \
   org.label-schema.name="ownCloud PHP" \
@@ -34,7 +34,6 @@ RUN apt-get update -y && \
     php-apcu \
     php-redis \
     php-smbclient \
-    sernet-samba-client \
     patch \
     mysql-client \
     postgresql-client \
```

I could do:
```sh
curl -L https://github.com/owncloud/user_ldap/releases/download/v0.13.0/user_ldap-0.13.0.tar.gz > server/user_ldap.tar.gz
curl -L https://download.owncloud.org/community/owncloud-10.2.1.tar.bz2 > server/owncloud-10.2.1.tar.bz2
for REPO in ubuntu php base server
do
  cd $REPO && docker build -t owncloud/$REPO:vytas . && cd ..
done

cat << EOF >| .env
OWNCLOUD_VERSION=vytas
OWNCLOUD_DOMAIN=localhost
ADMIN_USERNAME=admin
ADMIN_PASSWORD=admin
HTTP_PORT=80
EOF

wget -O docker-compose.yml https://raw.githubusercontent.com/owncloud-docker/server/master/docker-compose.yml

docker-compose up -d
```

In the end, my move from `x86_64` to `aarch64` was successful! Would have been perfect, if build process of Owncloud docker images supported arm64. Looks like there are no blockers! Well, maybe except the `sernet-samba-client` thing.

**Comment by tboerger** [comment]
I know how to build these images, but currently our ci system doesn't provide arm32/arm64 agents, so this is our major blocker so far.

Beside that, the sernet samba package is also a hard requirement for the docker images.

**Comment by tboerger** [comment]
Finally we are building arm and arm64 Docker images now 🎉

**Comment by loudjanilef** [comment]
Hello,
I am sorry to comment on this closed issue but I am facing a similar problem.
I'm trying to use owncloud/server on a Raspberry Pi 4 but it lookslike there is no linux/arm/v7 version of the image.
Would it be possible to add one ?

**Comment by michaelstingl** [comment]
Looks like latest 10.6 versions were only build in `arm64v8`:
https://hub.docker.com/r/owncloud/server/tags?page=1&ordering=last_updated&name=arm

Only old versions available in `v7`:
https://hub.docker.com/r/owncloud/server/tags?page=1&ordering=last_updated&name=v7

**Comment by loudjanilef** [comment]
Thank you for your help, i could install version 10.4.1 on my raspberry.
Is there any reason why arm32v7 images are discontinued ?

**Comment by michaelstingl** [comment]
I think 32-bit is no longer supported.

Wikipedia says Raspberry Pi 4 Mod. B runs on Cortex-A72, and this is Armv8 design.

**Comment by michaelstingl** [comment]
CI can't build arm32v7, PR was submitted, but it's not merged yet…

Related:
- https://github.com/owncloud-docker/ubuntu/pull/44
- https://github.com/drone-plugins/drone-docker/pull/312

**Comment by wkloucek** [comment]
> Thank you for your help, i could install version 10.4.1 on my raspberry.
> Is there any reason why arm32v7 images are discontinued ?

They are not discontinued, but temporally unavailable because of a Seccomp problem in Ubuntu / Docker: https://github.com/moby/moby/issues/40734

To get this image back again we need get https://github.com/drone-plugins/drone-docker/pull/312 merged and then merge a row of PRs beginning here https://github.com/owncloud-docker/ubuntu/pull/44.

**Comment by loudjanilef** [comment]
> Wikipedia says Raspberry Pi 4 Mod. B runs on Cortex-A72, and this is Armv8 design.
Raspberry Pi 4B has a ArmV8 core but raspbian still use a 32 bits kernel on stable version.

> To get this image back again we need get drone-plugins/drone-docker#312 merged and then merge a row of PRs beginning here owncloud-docker/ubuntu#44.
Thank you for the update, I'll keep these issues in sight.

**Comment by wkloucek** [comment]
@loudjanilef arm32v7 image is now available for 10.6.0

**Comment by loudjanilef** [comment]
Hi,
Sorry to bring this issue up
The arm32v7 image isn't available anymore for versions after 10.9
Is it on purpose ? Is there any alternative repo where is can find it ?

**Comment by michaelstingl** [comment]
32-bit ARM no longer possible because of upstream problems:
- https://github.com/owncloud-docker/server/pull/358
